master/internal/api_workspace.go

```diff
@@ -521,6 +554,117 @@ func (a *apiServer) PatchWorkspace(
 		errors.Wrapf(err, "error refetching updated workspace (%d) from db", currWorkspace.Id)
 }
 
+func (a *apiServer) addWorkspaceNamespaceBinding(ctx context.Context,
+	req *apiv1.AddWorkspaceNamespaceBindingRequest, tx *bun.Tx, wkspID int,
+) (*apiv1.AddWorkspaceNamespaceBindingResponse, error) {
+	// TODO(amanda): When we land auto-creating namespaces, add an AND to this if condition checking
+	// for nil namespace name AND False auto-create namespace.
+	// NamespaceName is an optional param in the request's proto message to leave room for the
+	// auto-create namespace feature.
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := workspace.AuthZProvider.Get().
+		CanAddWorkspaceNamespaceBindings(ctx, *curUser, int32(wkspID)); err != nil {
+		return nil, status.Error(codes.PermissionDenied, err.Error())
+	}
+
+	err = a.validateRequestClusterName(req.ClusterName)
+	if err != nil {
+		return nil, err
+	}
+
+	// Verify that the namespace exists in the Kubernetes cluster for the corresponding RM.
+	err = a.m.rm.VerifyNamespaceExists(*req.NamespaceName, req.ClusterName)
+	if err != nil {
+		return nil, fmt.Errorf("error verifying Kubernetes namespace: %w", err)
+	}
+
+	if req.ClusterName == nil {
+		emptyClusterName := ""
+		req.ClusterName = &emptyClusterName
+	}
+
+	// Change the workspace-namespace binding for the given cluster if it already exists. If no
+	// namespace is bound to the workspace for the specified cluster, add one.
+	findWsNsQuery := tx.NewSelect().Model(&model.WorkspaceNamespace{}).
+		Where("workspace_id = ?", wkspID).
+		Where("cluster_name = ?", req.ClusterName)
+
+	var wsns model.WorkspaceNamespace
+	err = findWsNsQuery.Scan(ctx, &wsns)
+	if err != nil {
+		if err != sql.ErrNoRows {
+			return nil, fmt.Errorf("error getting the current workspace-namespace binding: %w", err)
+		}
+		// The workspace has no namespace binding for the specified cluster, so we add one.
+		wsns = model.WorkspaceNamespace{
+			NamespaceName: *req.NamespaceName,
+			ClusterName:   *req.ClusterName,
+			WorkspaceID:   wkspID,
+		}
+		err = tx.NewInsert().Model(&wsns).Scan(ctx)
```

Comment:
this should be Exec instead of Scan. Scan will give an error saying there are no rows in result set